spec/lib/extensions/ar_virtual_spec.rb

```diff
@@ -33,12 +33,22 @@ def self.connection
         end
         belongs_to :ref1, :class_name => 'TestClass', :foreign_key => :col1
       end
+
+      class TestClassWithDeprecatedAttr < TestClassBase
+        self.table_name = :test_classes
+        include DeprecationMixin
+        def self.connection
+          TestClassBase.connection
+        end
+        deprecate_attribute :old_col1, :col1
```

Comment:
Did you consider adding a new, deprecated column to the existing `TestClass`? Alternatively, you could add a separate context for the test you added below with its own before/after hooks to manage the lifecycle of this fixture, so that it's not setup/torn down for a lot of tests that don't use it

Reply:
:+1: I think you could just add this to an existing test class